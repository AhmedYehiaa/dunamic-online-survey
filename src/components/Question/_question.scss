.question {
  font-size: 2.4rem;
  font-weight: 400;
  margin-bottom: 50px;
  max-width: 75%;
  margin: 0 auto 40px;
  text-align: left;
}