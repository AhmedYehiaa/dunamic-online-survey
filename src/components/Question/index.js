import React from 'react';
import PropTypes from 'prop-types';

const Question = ({ text }) => (
  <h1 className="question">{text}</h1>
);

Question.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Question;
