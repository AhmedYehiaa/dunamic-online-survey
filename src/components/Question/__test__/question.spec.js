import React from 'react';
import Question from '../index';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Question text="http://www.facebook.com" />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
