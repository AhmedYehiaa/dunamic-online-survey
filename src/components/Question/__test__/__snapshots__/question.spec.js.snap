// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<h1
  className="question"
>
  http://www.facebook.com
</h1>
`;
